frontend/src/hooks/useSearch.ts

```diff
@@ -1,85 +1,94 @@
+ * @param initialSort Any sort parameters to apply to the filter.
+ * @param initialPage The initial paginated page for this filter.
+ * @param initialPageSize The initial page size for this filter.
  */
-export function useSearch(
-  initialFilter: ILeaseFilter,
-  initialSort: TableSort<ILeaseSearchResult> = {},
+export function useSearch<ISearchResult extends object, IFilter extends object>(
+  initialFilter: IFilter,
+  apiCall: (
+    params: IPaginateRequest<IFilter>,
+  ) => Promise<AxiosResponse<IPagedItems<ISearchResult>>>,
+  initialSort: TableSort<ISearchResult> = {},
   initialPage = 0,
   initialPageSize = 10,
-): SearchState {
+): SearchState<IFilter, ISearchResult> {
   // search state
   const [error, setError] = useState<Error>();
-  const [filter, setFilter] = useState<ILeaseFilter>(initialFilter);
-  const [sort, setSort] = useState<TableSort<ILeaseSearchResult>>(initialSort);
+  const [filter, setFilter] = useState<IFilter>(initialFilter);
+  const [sort, setSort] = useState<TableSort<ISearchResult>>(initialSort);
   const [currentPage, setCurrentPage] = useState(initialPage);
   const [pageSize, setPageSize] = useState(initialPageSize);
-  const [results, setResults] = useState<ILeaseSearchResult[]>([]);
+  const [results, setResults] = useState<ISearchResult[]>([]);
   const [totalItems, setTotalItems] = useState(0);
   const [totalPages, setTotalPages] = useState(0);
+  const [loading, setLoading] = useState(false);
   // manual refresh
   const [refreshIndex, setRefreshIndex] = useState(0);
-  // api helper
-  const searchFn = useFetcher();
+  const searchFn = useFetcher<ISearchResult, IFilter>(apiCall);
   // is this component/hook mounted?
   const isMounted = useIsMounted();
 
-  const setSearchOutput = useCallback(
-    (apiResponse?: IPagedItems<ILeaseSearchResult>, pageSize = 10) => {
-      if (apiResponse?.items) {
-        setResults(apiResponse.items);
-        setTotalItems(apiResponse.total);
-        setTotalPages(Math.ceil(apiResponse.total / pageSize));
-      } else {
-        setResults([]);
-        setTotalItems(0);
-        setTotalPages(0);
-      }
-    },
-    [],
-  );
+  const setSearchOutput = useCallback((apiResponse?: IPagedItems<ISearchResult>, pageSize = 10) => {
+    if (apiResponse?.items) {
+      setResults(apiResponse.items);
+      setTotalItems(apiResponse.total);
+      setTotalPages(Math.ceil(apiResponse.total / pageSize));
+    } else {
+      setResults([]);
+      setTotalItems(0);
+      setTotalPages(0);
+    }
+  }, []);
 
   // update search results whenever new data comes back from API endpoints
   useDeepCompareEffect(() => {
     if (!filter) return;
 
     async function callApi() {
       try {
+        setLoading(true);
         const { data } = await searchFn(filter, sort, currentPage, pageSize);
         if (isMounted()) {
-          setSearchOutput(data);
+          setSearchOutput(data, pageSize);
```

Comment:
fix bug where the total number of pages wasn't being updated after the page size is changed.

Reply:
> Should the search section stretch to fill the top bar, pushing the button to align with the page? It looks a tad odd to have the search component not filling all its space ![image](https://user-images.githubusercontent.com/12201292/140990086-6d1631c2-fbca-47cb-9277-c06176911273.png)

I'll run that by Bobbi, but the screenshot matches the mockup.

Reply:
Answer was that it shouldn't fill all the space, but the button should center itself in whatever space is left.